corehq/apps/linked_domain/static/linked_domain/js/domain_links.js

```diff
@@ -236,12 +245,70 @@ hqDefine("linked_domain/js/domain_links", [
         self.modelsToPush = ko.observableArray();
         self.buildAppsOnPush = ko.observable(false);
         self.pushInProgress = ko.observable(false);
+        self.shouldShowSelectedERMDomain = ko.observable(false);
+        self.shouldShowSelectedMRMDomain = ko.observable(false);
         self.enablePushButton = ko.computed(function () {
             return self.domainsToPush().length && self.modelsToPush().length && !self.pushInProgress();
         });
+        self.containsLiteAndFullLinks = ko.computed(function () {
+            if (!self.parent.hasFullAccess) {
+                // should not contain both
+                return false;
+            }
+            const accessSet = new Set();
+            for (const link of Object.values(self.parent.domainLinksByNames())) {
+                // if any link is limited, contains both
+                accessSet.add(link.hasFullAccess);
+                if (accessSet.size >= 2) {
+                    return true;
+                }
+            }
+            return false;
+        });
+
+        self.domainsToPushSubscription = self.domainsToPush.subscribe(function (newValue) {
+            // receives updates every time a domain is selected/unselected from the multiselect
+
+            // handles the Add All edge case if both lite and full access links exist
+            if (newValue.length > 1 && self.containsLiteAndFullLinks()) {
+                // the non-enterprise domains would have already been hidden in the callback, so just show the info text
+                self.shouldShowSelectedERMDomain(true);
+                // no need to rebuild multiselect
+                return;
+            }
+
+            if (newValue.length > 0) {
+                var selectedDomainLink = self.parent.domainLinksByNames()[newValue[0]];
+                var pushedNonEnterpriseLink = !selectedDomainLink.hasFullAccess;
+                for (const option of $('#domain-multiselect')[0].options) {
```

Comment:
```suggestion
                $('#domain-multiselect option').each(function (option) {
```
Nitpick/style thought. I like that it has slightly less punctuation.

Reply:
Yeah I think we briefly chatted about this before, but `$('#domain-multiselect')[0].options` returns an `HTMLCollection` which is iterable but does not have support for `forEach`, so we would first have to use `Array.from(...)`. So our options are:
```
for (const option of $('#domain-multiselect')[0].options) {...}
```
```
Array.from($('#domain-multiselect')[0].options).forEach(function (option) {...});
```

So that's why I stuck with the first option, but if there is still a compelling reason to use forEach I'm open to it. Either way, not pretty.

Reply:
I do remember chatting about options now - nevermind, no change to make here.